ext/pdo/pdo_stmt.c

```diff
@@ -1298,11 +1242,11 @@ PHP_METHOD(PDOStatement, fetchAll)
 			}
 	}
 
-
 	if (fetch_mode == PDO_FETCH_USE_DEFAULT) {
 		flags |= stmt->default_fetch_type & PDO_FETCH_FLAGS;
 		fetch_mode = stmt->default_fetch_type & ~PDO_FETCH_FLAGS;
 		how = fetch_mode | flags;
+		is_default_fetch_mode = true;
```

Comment:
At first I didn't get why this variable was here, but I see it now: fetch_mode is overwritten so the check below would be incorrect if this variable was not there...

Reply:
Turns out not needed.